docs/book/src/tasks/experimental-features/cluster-class-operations.md

```diff
@@ -22,11 +22,11 @@ Once the changes are applied, the topology controller reacts as described in the
 | Changed field                                   | Effects on Clusters                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
 |-------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | infrastructure.ref                              | Corresponding InfrastructureCluster objects are updated (in place update).                                                                                                                                                                                                                                                                                                                                                                                                                     |
-| controlPlane.metadata                           | If labels/annotations are added or their value is changed, ControlPlane objects are updated (in place update) and corresponding controlPlane Machines are updated (rollout).  <br /> <br />If labels/annotations are deleted, the label is downgraded from managed to instance specific value, but no changes are applied.  <br /> In order to actually remove the label it is required for the user to change ControlPlane object in place.                                                                             |
+| controlPlane.metadata                           | If labels/annotations are added, changed or deleted the ControlPlane objects are updated (in place update).<br /><br /> In case of KCP, corresponding controlPlane Machines are updated (rollout) only when adding or changing labels or annotations; deleted label should be removed manually from machines or they will go away automatically at the next machine rotation.      |
```

Comment:
I wonder if we want to add something structured like https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/architecture/controllers/metadata-propagation.md#kubeadmcontrolplane so we can move them there once this is not experimental.

Reply:
same as above